### ISSUE TITLE
How does multi-head attention settings in train_rnn.yaml

### ISSUE THREAD

**Issue by jubang0219** [espnet/espnet#1799]
Hi there, 

I want to train multi-head attention in RNN architecture (not transformer architecture).
I would like to ask about the settings for using this.

In the "espnet/egs/wsj/asr1/conf/tuning/train_rnn.yaml", I found a config-file to use the multi-head attention with "atype: location" and "aheads: 4" settings.

_# attention related
atype: location
adim: 320
awin: 5
aheads: 4
aconv-chans: 10
aconv-filts: 100_

On the other hand, in the "espnet/espnet/nets/pytorch_backend/rnn/attentions.py", I also found the "atype: multi_head_loc" and "aheads: 4" settings.

In summary, I want to use multi-head attention (_location_ or _additive_ attention versions with 4 heads). Which config should I use, multi_head_add or add?

Thanks!

**Comment by sw005320** [comment]
I think it should be `multi_head_add` but @kan-bayashi, could you clarify it?
It's actually confusing and some configs use `aheads` for non-multi-head attention modes.
It should be fixed and it would also be better to clarify it in the help message or documentation???

**Comment by jubang0219** [comment]
Thanks S. Watanabe.

I have one more question.

In RNN, when using "adim:1024, aheads:8" settings, each head has a "1024" dimension. (Total 8*1024=#heads*adim)
(they are calculated by #aheads*adim on Class AttMultiHeadAdd)
https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/rnn/attentions.py

On the other hand, in transformer, when using "adim:1024, aheads:8", each head has a "128(=1024//8)" dimension.
(they are automatically divided by #aheads on Class MultiHeadedAttention)
https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/transformer/attention.py

Why are the "adim" per head differences between RNN and transformer architecture?

In the RNN architecture, is this a fair comparison if I set: (aheads=1,adim=1024), (aheads=2,adim=512), (aheads=4,adim=256)

Thanks!

**Comment by sw005320** [comment]
Yes, they are different because we implement them independently (and transformer one refers to the other existing implementations).
Sorry for the confusion.

**Comment by jubang0219** [comment]
Thank you for the reply.
